runhouse/resources/serverless/aws_lambda_function.py

```diff
@@ -87,7 +88,7 @@ def __init__(
         """
         if isinstance(fn, Callable):
```

Comment:
can remove all these checks here since all this logic is in the factory

Reply:
did some refactor, since not all logic was implemented in the factory method.